Documentation/Books/Manual/Programs/Arangodump/Maskings.md

```diff
@@ -0,0 +1,68 @@
+Arangodump Data Maskings
+========================
+
+*--maskings path-of-config*
+
+It is possible to mask certain fields during dump. A JSON config file is
+used to define with fields should be masked and how.
+
+The general structure of the config file is
+
+    {
+      "collection-name": {
+        "type": MASKING_TYPE
+        "maskings" : [
+          MASKING1,
+          MASKING2,
+          ...
+        ]
+      },
+      ...
+    }
+
+## MASKING_TYPE
+
+This is a string describing how to mask this collection. Possible values are
+
+- "ignore": the collection is ignored complete and not even the sturcture data
+  is dumped.
+
+- "structure": only the collection structure is dumped, but no data at all
+
+- "full": the collection structure and all data is dumped. However, the data
+  is subject to maskings defined in the attribute maskinks.
+
+For example:
+
+    {
+      "private": {
+        "type": "ignore"
+      },
+
+      "log": {
+        "type": "structure"
+      },
+
+      "person": {
+        "type": "full",
+        "maskings": [
+          {
+            "path": "name",
```

Comment:
What about nested attributes? Will `my.nested.attribute` work here?

In View definitions for ArangoSearch, we use a nested configuration format. The benefit is that it's possible to have an attribute `{ "my.attribute": ... }` but also `{ "my": { "attribute": ... } }` and clearly define which one to index.

For bind parameters, we support arrays like `["my", "attribute"]`. However, in index definitions (and simple queries e.g. `byExample()`) we don't support attributes with dots in their names.

Old report about this for byExample():
https://trello.com/c/6WBsCnht/1142-dot-still-magic-character-in-attribute-keys-in-byexample

Reply:
@Simran-B isn't there a convention to escape attribute names in AQL?

Reply:
You can escape e.g. language keywords in AQL yes, but there it's not a problem anyway, because you can do `doc.@bindparam` with `{ "bindparam": ["my", "attribute"] }`. There is no such mechanism for fields to index, in the JS API / Simple Queries like byExample() and also the web interface.
I created a separate issue for this: https://github.com/arangodb/planning/issues/3240

Reply:
ok, for the dump definition I would prefer the quoting

Reply:
Not sure if I follow. The following:

```json
{ "path": "`my.attribute`" }
```

would only shift the problem to backticks. It's unlikely that someone will put backticks in the name of an attribute key, but it's allowed. For now, it would be fine to interpret the dots path separator I guess, but we should keep this in mind for 4.0 and fix it there. An array would work:

```json
{ "path": ["my", "attribute"] } 
```

If multiple paths are allowed, then we should require an outer array (otherwise it will be ambiguous again):

```json
{ "paths": [ ["my", "attribute"] ] } 
```